File: 2020-CompPhys/Homework_solutions/HW1_2/problem 1/problem1.m
xvals=zeros(1,21);
yvals=zeros(1,21);
dt=1;
for iter=0:20
   dt=10^(-iter);
   diff=difference(dt);
   xvals(iter+1)=dt;
   yvals(iter+1)=diff;
end

f=figure;
hold all;
ax=gca;
plot(xvals,yvals,'-o','color','black');
ax.XScale='log';
ax.YScale='log';
title('matlab');
xlabel('dt');
ylabel('difference')
box on;
set(f,'Visible','off');
saveas(f,'matlab','epsc');
hold off


function g = difference(dt)
    e=9*exp(-2);
    f=1/(1+e);
    der=e*f*f;
    fnew=1/(1+9*exp(-2-dt));
    g=abs(der-(fnew-f)/dt);
end

File: 2020-CompPhys/Homework_solutions/HW1_2/problem 1/problem1.py
import numpy as np
import math
import matplotlib as mp
import matplotlib.pyplot as plt

def difference(dt):
	e=9*math.exp(-2)
	f=1/(1+e)
	der=e*f*f
	fnew=1/(1+9*math.exp(-2-dt))
	return abs(der-(fnew-f)/dt)


xvals=[0]*21
yvals=[0]*21
for iter in range(21):
	dt=10**(-iter);
	xvals[iter]=dt;
	yvals[iter]=difference(dt);


# for your solutions, you can export data and plot it in matlab if you want.  
# this gives a sense of hwo to do it it python.

plt.scatter(xvals,yvals,color='k')
ax=plt.gca()
ax.set_xscale('log')
ax.set_yscale('log')
plt.title('python error')
plt.xlabel('dt')
plt.ylabel('difference')

plt.loglog(xvals,yvals,color='k')
plt.savefig('python.eps',format='eps')

plt.clf()


# plot the results of the c++ data too, which are stored in a text file
data=np.genfromtxt('c_problem1.txt',delimiter=',')

plt.scatter(xvals,yvals,color='m',marker=6)
ax=plt.gca()
ax.set_xscale('log')
ax.set_yscale('log')
plt.title('c++ error')
plt.xlabel('dt')
plt.ylabel('difference')

plt.loglog(xvals,yvals,color='m')
plt.savefig('cpp.eps',format='eps')

plt.clf()



